Actions/ReadSettings/ReadSettings.ps1

```diff
@@ -3,99 +3,81 @@ Param(
-                $settings.appBuild = $settings.runNumberOffset + [Int32]($ENV:GITHUB_RUN_NUMBER)
-                $settings.appRevision = [Int32]($ENV:GITHUB_RUN_ATTEMPT) - 1
-            }
-            1 { # Use RUN_ID and RUN_ATTEMPT
-                OutputError -message "Versioning strategy 1 is no longer supported"
-            }
-            2 { # USE DATETIME
-                $settings.appBuild = [Int32]([DateTime]::UtcNow.ToString('yyyyMMdd'))
-                $settings.appRevision = [Int32]([DateTime]::UtcNow.ToString('HHmmss'))
-            }
-            15 { # Use maxValue
-                $settings.appBuild = [Int32]::MaxValue
-                $settings.appRevision = 0
-            }
-            default {
-                OutputError -message "Unknown version strategy $versionStrategy"
-                exit
-            }
+if ($settings.versioningstrategy -ne -1) {
+    switch ($settings.versioningStrategy -band 15) {
+        0 { # Use RUN_NUMBER and RUN_ATTEMPT
+            $settings.appBuild = $settings.runNumberOffset + [Int32]($ENV:GITHUB_RUN_NUMBER)
+            $settings.appRevision = [Int32]($ENV:GITHUB_RUN_ATTEMPT) - 1
+        }
+        1 { # Use RUN_ID and RUN_ATTEMPT
+            OutputError -message "Versioning strategy 1 is no longer supported"
+        }
+        2 { # USE DATETIME
+            $settings.appBuild = [Int32]([DateTime]::UtcNow.ToString('yyyyMMdd'))
+            $settings.appRevision = [Int32]([DateTime]::UtcNow.ToString('HHmmss'))
+        }
+        15 { # Use maxValue
+            $settings.appBuild = [Int32]::MaxValue
+            $settings.appRevision = 0
+        }
+        default {
+            OutputError -message "Unknown version strategy $versionStrategy"
+            exit
         }
     }
+}
 
-    $outSettings = @{}
-    $settings.Keys | ForEach-Object {
-        $setting = $_
-        $settingValue = $settings."$setting"
-        $outSettings += @{ "$setting" = $settingValue }
-        if ($getSettings -contains $setting) {
-            if ($settingValue -is [System.Collections.Specialized.OrderedDictionary] -or $settingValue -is [hashtable]) {
-                Add-Content -Encoding UTF8 -Path $env:GITHUB_ENV -Value "$setting=$(ConvertTo-Json $settingValue -Depth 99 -Compress)"
-            }
-            elseif ($settingValue -is [String] -and ($settingValue.contains("`n") -or $settingValue.contains("`r"))) {
-                throw "Setting $setting contains line breaks, which is not supported"
-            }
-            else {
-                Add-Content -Encoding UTF8 -Path $env:GITHUB_ENV -Value "$setting=$settingValue"
-            }
+$outSettings = @{}
+$settings.Keys | ForEach-Object {
+    $setting = $_
+    $settingValue = $settings."$setting"
+    $outSettings += @{ "$setting" = $settingValue }
+    if ($getSettings -contains $setting) {
+        if ($settingValue -is [System.Collections.Specialized.OrderedDictionary] -or $settingValue -is [hashtable]) {
```

Comment:
Minor: what if the setting's value is a hashtable that contains a string that contains a new line?

Reply:
ConvertTo-Json encodes line breaks:

PS C:\oldtemp> "$(ConvertTo-Json $s -Depth 99 -Compress)"
{"test":"test\r\ntest"}

Reply:
But it does that for strings as well:

```
>> "test`r`ntest" | ConvertTo-Json -Depth 99 -Compress
>> "test\r\ntest"
```

So, what exactly is the issue with new lines?

Reply:
The problem with newlines in settings was from back, when we had all settings as individual environment variables.
And it will also be when we request a setting as environment var (get)
We could remove this check for any setting which is not part of get.
Not sure it should be part of this PR though.

Reply:
I agree that it's not part of this PR. Just caught my eye.